packages/grpc-js/src/server-call.ts

```diff
@@ -553,102 +553,100 @@ export class Http2ServerCallStream<
     return metadata;
   }
 
-  receiveUnaryMessage(
-    encoding: string,
-    next: (
-      err: Partial<ServerStatusResponse> | null,
-      request?: RequestType
-    ) => void
-  ): void {
-    const { stream } = this;
+  receiveUnaryMessage(encoding: string): Promise<RequestType | void> {
+    return new Promise((resolve, reject) => {
+      const { stream } = this;
+
+      let receivedLength = 0;
+
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const call = this;
+      const body: Buffer[] = [];
+      const limit = this.maxReceiveMessageSize;
 
-    let receivedLength = 0;
+      this.stream.on('data', onData);
+      this.stream.on('end', onEnd);
+      this.stream.on('error', onEnd);
 
-    // eslint-disable-next-line @typescript-eslint/no-this-alias
-    const call = this;
-    const body: Buffer[] = [];
-    const limit = this.maxReceiveMessageSize;
+      async function onData(chunk: Buffer) {
```

Comment:
Making a function `async` has two main effects: it can `await` things and it returns a promise. This function does not `await` anything and it is used as an event listener so its return value is not used. So, I don't see any reason to make that change.

`onEnd` has the same issue.

Reply:
Sorry, this was a slight oversight. I halfway through opted to make this change as little as possible and overlooked this